### ISSUE TITLE
Don't store users phone numbers on-chain

### ISSUE THREAD

**Issue by avive** [karma-coin/karmachain#121]
# Motivation
The biggest privacy concern around Karmachain is storage of users phone numbers. We'd like to keep all existing features, including sending a text message to a phone number of a person who doesn't have a Karma Coin account and modify Karmachain so it stores hashes of phone numbers instead of plain-text number.

# Solution Outline
- Agree on a common and robust hash function to be used by clients and karmachain. e.g. SHA512.
- Transactions will include a hash of phone numbers instead of numbers. Hashing done on client before signing txs.
- Change all indexes and store to work with phone number hashes.
- Change API to return on-chain account for a phone number by hashing it and comparing the hashes with the indexed accounts by hash.
- Deploy a web service that receives the appreciation and the receiver's phone number and sends the sms message based on it to that number about the appreciation. 
- Client uses the chain API to check if the appreciation's receiver phone number is on-chain. If it is note, it calls a new web-service with information about the tx and the phone number. The service will send an sms to the phone number about the appreciation.

**Comment by avive** [comment]
I think we should use the modern Sha384 instead of SHA512.

**Comment by avive** [comment]
I've implemented as reference VerifyNumberEx that returns verification evidence with phone number hash instead of phone number. Here: https://github.com/karma-coin/karmacoin-server/blob/main/crates/server/src/services/verifier/verify_number_ex.rs
Few notes:
1. No need to change the auth dart server. It can work without modifications - accepting an AuthRequest from the verifier.
2. Request to verifier includes client's phone number as it is on a secure channel and not stored on chain.
3. Verifier returns signed evidence includes a blake2 hash of the number to include in signup tx.
4. Note that the structure of the evidence changed - evidence binary (serielized protobuf message) is signed by verifier for enhanced compatibly across protobufs implementations. All changes reflected in protobuf defs...
5. Added support for bypass token for testing purposes - if it is included in the call for verifier then it returns evidence without using the dart auth service - assuming that the account is associated with the phone number.
@HolyGrease
